flink-connector-mysql-cdc/src/main/java/com/alibaba/ververica/cdc/connectors/mysql/MySqlValidator.java

```diff
@@ -0,0 +1,135 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.alibaba.ververica.cdc.connectors.mysql;
+
+import org.apache.flink.table.api.TableException;
+
+import com.alibaba.ververica.cdc.debezium.Validator;
+import io.debezium.config.Configuration;
+import io.debezium.connector.mysql.MySqlConnection;
+import io.debezium.connector.mysql.MySqlConnectorConfig;
+
+import java.io.Serializable;
+import java.sql.SQLException;
+import java.util.Arrays;
+import java.util.Properties;
+
+import static io.debezium.connector.mysql.MySqlConnection.MySqlConnectionConfiguration;
+
+/**
+ * The validator for MySQL: it only cares about the version of the database is larger than or equal
+ * to 5.7. It also requires the binlog format in the database is ROW.
+ *
+ * <p>We adapt part of the logic from the Debzium to validate in the compile phase rather than in
+ * runtime time.
+ */
+public class MySqlValidator implements Validator, Serializable {
+
+    private static final long serialVersionUID = 1L;
+
+    private final Configuration configuration;
+
+    public MySqlValidator(Properties properties) {
+        this(Configuration.from(properties));
+    }
+
+    public MySqlValidator(Configuration configuration) {
+        this.configuration = configuration;
+    }
+
+    @Override
+    public void validate() {
+        try (MySqlConnection connection =
+                new MySqlConnection(new MySqlConnectionConfiguration(configuration))) {
+            if (!isVersionSatisfied(connection)) {
+                throw new TableException(
+                        "Currently Flink MySql CDC connector only supports MySql whose version is larger or equal to 5.7.");
+            }
+
+            MySqlConnectorConfig mySqlConnectorConfig = new MySqlConnectorConfig(configuration);
+            if (mySqlConnectorConfig.getSnapshotMode().shouldStream()) {
```

Comment:
We can  use flink's option here rather than debezium's， because we want to debezium free as a long goal